### PR TITLE
Gas miner power

### PR DESCRIPTION
**Pull request by bloons3**
## About The Pull Request
- makes atmos miners require power (currently they are in space ruin mode and consume no power)
- makes atmos miners have a lower pressure limit
- makes atmos miners create less gas per second

## Changelog
:cl:
tweak: gas miners on maps are no longer in space ruin mode and actively consume power
/:cl:

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by bloons3** [comment]
Numbers subject to change

**Comment by Archanial** [comment]
Better solution would be to make a subtype of miner that consumes power instead of manually editing it on maps.

**Comment by utilbird** [comment]
I agree with Archanial

**Comment by OceanFish1** [comment]
This should apply to station maps, not space ruins.

**Comment by Cartlord** [comment]
...But why? What's the gain from making breathable air less reliable?

**Comment by ghost** [comment]
> ...But why? What's the gain from making breathable air less reliable?

This does not make air less reliable, if atmospherics does not have power you would not get air in the first place.

**Review by ghost — DISMISSED** [review]

**Comment by bloons3** [comment]
> This should apply to station maps, not space ruins.

This only applies to the miners on the station, which cannot be crafted.

**Comment by PowerfulBacon** [comment]
ew changing vars in mapping

**Comment by OceanFish1** [comment]
> > This should apply to station maps, not space ruins.
> 
> This only applies to the miners on the station, which cannot be crafted.

understandable, have a great day

**Comment by FortSumter** [comment]
Make the gas miners gas output related to the amount of power the reciever. This could finally provide a reason to do the advanced sm setup to produce the max amount of power.

**Review by utilbird — CHANGES_REQUESTED** [review]
Make these a separate subtype rather than var editing the maps

**Comment by bloons3** [comment]
Requesting a code review from @TheFakeElon 
This is DNM until mapping changes are made to add power cables to the miners.

**Comment by PowerfulBacon** [comment]
DNM at author request.
As a side note, what is the point of this if you can't actually turn them off and they are default station things.

**Comment by utilbird** [comment]
@francinum ?

**Comment by bloons3** [comment]
> DNM at author request.
> As a side note, what is the point of this if you can't actually turn them off and they are default station things.

They turn off when they reach their pressure limit, which is reduced.

The reason for requiring power is that they generate infinite resources even when the station is completely destroyed because they require no power at all.

**Comment by bloons3** [comment]
Not DNM, doesn't require power cables

**Comment by bloons3** [comment]
Looking at testmerge on Sage:

- Gas miners don't draw a ton of power
- They do turn off when the area APC turns off Environment
- They don't make crazy amounts of gas, but you could run the incinerator indefinitely, and definitely a CO2 SM where the CO2 is consumed over the long term
- the bug description is showing for some reason

**Review by bloons3 — COMMENTED** [review]
reviewed

File: code/modules/atmospherics/machinery/other/miner.dm [hunk 1]
```diff
@@ -5,9 +5,11 @@
 #define GASMINER_POWER_KPA 3
 #define GASMINER_POWER_FULLSCALE 4
 
+#define GASMINER_DESC "Gasses mined from the gas giant below (above?) flow out through this massive vent."
+
 /obj/machinery/atmospherics/miner
 	name = "gas miner"
-	desc = "Gasses mined from the gas giant below (above?) flow out through this massive vent."
+	desc = GASMINER_DESC
```
Comment (PowerfulBacon) [inline code/modules/atmospherics/machinery/other/miner.dm]: ???

File: code/modules/atmospherics/machinery/other/miner.dm [hunk 2]
```diff
@@ -175,3 +177,60 @@
 	name = "\improper Water Vapor Gas Miner"
 	overlay_color = "#99928E"
 	spawn_id = /datum/gas/water_vapor
+
+/obj/machinery/atmospherics/miner/station
+	power_draw = GASMINER_POWER_FULLSCALE
+	spawn_mol = MOLES_CELLSTANDARD / 10
+	desc = "This item should not exist, file a bug report if you see this message"
```
Comment (PowerfulBacon) [inline code/modules/atmospherics/machinery/other/miner.dm]: ???

**Review by PowerfulBacon — DISMISSED** [review]

File: code/modules/atmospherics/machinery/other/miner.dm (hunk 2 above)
Comment (bloons3) [reply code/modules/atmospherics/machinery/other/miner.dm]: Is there a better solution to only have the text description in one place?
The other ones are mapping items, and the `/obj/machinery/atmospherics/miner/station` should never be mapped in, only the subtypes of it.

Comment (PowerfulBacon) [reply code/modules/atmospherics/machinery/other/miner.dm]: well it won't work anyway, it doesnt need the desc like this

Comment (bloons3) [reply code/modules/atmospherics/machinery/other/miner.dm]: Visibly broken instead of silently

**Review by utilbird — DISMISSED** [review]
Should be good to go after bacon's review is addressed, imo either change the descriptions for each one so they say what gas they mine + a description define. Or just remove the overrides completely for the station subtype

**Comment by bloons3** [comment]
@TheFakeElon reee

**Comment by ike709** [comment]
Fails CI.

**Comment by bloons3** [comment]
> Fails CI.

not my fault, rerun tests

image

This is from an unrelated PR that modified master

**Review by ike709 — APPROVED** [review]

**Pull request merged**